src/components/productCard/ProductCard.jsx

```diff
@@ -45,7 +49,18 @@ const ProductCard = ({
   };
 
   const addToCartHandler = () => {
-    addToCart(data, cartDispatch);
+    try {
+      const itemIndex = cartState?.cartItems?.findIndex(
```

Comment:
you can modularize this further by creating a function to check item in cart true or false

Reply:
Ok, will try that